assets/js/modules/sign-in-with-google/components/common/AnyoneCanRegisterDisabledNotice.js

```diff
@@ -0,0 +1,130 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement, Fragment } from '@wordpress/element';
+import { __, _x, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import {
+	CORE_USER,
+	PERMISSION_MANAGE_OPTIONS,
+} from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import {
+	BREAKPOINT_DESKTOP,
+	BREAKPOINT_XLARGE,
+	useBreakpoint,
+} from '../../../../hooks/useBreakpoint';
+import SettingsNotice, {
+	TYPE_INFO,
+} from '../../../../components/SettingsNotice';
+import Link from '../../../../components/Link';
+import InfoIcon from '../../../../../svg/icons/info-circle.svg';
+
+const ANYONE_CAN_REGISTER_DISABLED_NOTICE =
+	'sign-in-with-google-anyone-can-register-notice';
+
+export default function AnyoneCanRegisterDisabledNotice( { className } ) {
+	const breakpoint = useBreakpoint();
+
+	const canManageOptions = useSelect( ( select ) =>
+		select( CORE_USER ).hasCapability( PERMISSION_MANAGE_OPTIONS )
+	);
+	const isMultisite = useSelect( ( select ) =>
+		select( CORE_SITE ).isMultisite()
+	);
+	const generalSettingsURL = useSelect(
+		( select ) =>
+			new URL(
+				isMultisite ? 'network/settings.php' : 'options-general.php',
+				select( CORE_SITE ).getAdminURL()
+			).href
+	);
+
+	const anyoneCanRegister = useSelect( ( select ) =>
+		select( CORE_SITE ).getAnyoneCanRegister()
+	);
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			ANYONE_CAN_REGISTER_DISABLED_NOTICE
+		)
+	);
+
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	if ( isDismissed === true || anyoneCanRegister === true ) {
+		return null;
+	}
+
+	return (
+		<SettingsNotice
+			className={ classnames(
+				'googlesitekit-anyone-can-register-disabled-notice',
+				className
+			) }
+			type={ TYPE_INFO }
+			Icon={ InfoIcon }
+			dismiss
+			dismissCallback={ () =>
+				dismissItem( ANYONE_CAN_REGISTER_DISABLED_NOTICE )
+			}
+			dismissLabel={ __( 'Got it', 'google-site-kit' ) }
+			notice={ createInterpolateElement(
+				sprintf(
+					/* translators: %1$s: Setting name, %2$s: Sign in with Google service name */
+					__(
+						'Enable the %1$s setting to allow your visitors to create an account using the %2$s button. <br/>Visit <a>WP settings</a> page to manage this membership setting.',
+						'google-site-kit'
+					),
+					isMultisite
+						? __( '"Allow new registrations"', 'google-site-kit' )
+						: __( '"Anyone can register"', 'google-site-kit' ),
```

Comment:
These strings should use "smart quotes" eg. `“Allow new registrations“`/`“Anyone can register“`